overlay/overrides.nix

```diff
@@ -127,11 +127,10 @@ in rec {
     overrideAttrs = drv: {
       propagatedBuildInputs = drv.propagatedBuildInputs or [ ] ++ [
         pkgs.pkg-config
-      ] ++ lib.optional (pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform)
```

Comment:
Even if true... I don't see how removing this switch can make the result _more_ reliable.  The commit note is not exactly a strong reassurance :cat2:

Reply:
The issue is that `pkgs.stdenv.hostPlatform != pkgs.stdenv.buildPlatform` will never succeed because the `pkg-config` crate is a build dependency of the crate being cross-compiled. This means that this environment variable is never set and never propagated up the chain to crates that need it, e.g. `curl-sys`. Since this conditional isn't a reliable way of checking whether we are cross-compiling or not, and neither @trha and I are aware of any functional alternatives that work around this issue, we decided to set this variable always. I can't think of any reason where this would cause an issue.